src/overset/oversetAPI.F90

```diff
@@ -2237,51 +2238,80 @@ subroutine flagCellsInSurface(pts, npts, conn, nconn, flag, ncell, blockids, nbl
-                                    case (8)
-                                        coor(1:3) = x(i - 1, j, k, :)
-                                    end select
-
-                                    ! reset the "closest point already found" variable.
-                                    coor(4) = dStar
-                                    intInfo(3) = 0
-                                    call minDistancetreeSearchSinglePoint(ADT, coor, intInfo, &
-                                                                          uvw, dummy, 0, BB, &
-                                                                          frontLeaves, frontLeavesNew)
-                                    cellID = intInfo(3)
-                                    if (cellID > 0) then
-                                        ! Now check if this was successful or not:
-                                        if (checkInside()) then
-                                            ! Whoohoo! We are inside the region. Flag this cell
-                                            flag(cell_counter) = 1
-                                            exit
-                                        else
-                                            ! we are outside. now check if the projection distance is larger than
-                                            ! the max diagonal. if so, we can quit early here.
-                                            if (uvw(4) .gt. diag_max) then
-                                                ! projection is larger than our biggest diagonal.
-                                                ! other nodes wont be in the surface, so we can exit the cell early here
+                                ! First test the cell center
+                                coor(1:3) = eighth * ( &
+                                            x(i - 1, j - 1, k - 1, :) + &
+                                            x(i, j - 1, k - 1, :) + &
+                                            x(i - 1, j, k - 1, :) + &
+                                            x(i, j, k - 1, :) + &
+                                            x(i - 1, j - 1, k, :) + &
+                                            x(i, j - 1, k, :) + &
+                                            x(i - 1, j, k, :) + &
+                                            x(i, j, k, :))
+
+                                ! reset the "closest point already found" variable.
+                                coor(4) = dStar
+                                intInfo(3) = 0
+                                call minDistancetreeSearchSinglePoint(ADT, coor, intInfo, &
+                                                                      uvw, dummy, 0, BB, &
+                                                                      frontLeaves, frontLeavesNew)
+                                cellID = intInfo(3)
+
+                                keepChecking = .True.
+
+                                if (cellID > 0) then
```

Comment:
Current implementation is fine for now, but what I initial had in mind is instead of having these large if blocks with lots of code, it would also be possible to do something like the snippet below. The earlier if check `if (k_cgns .gt. k_min) then` could also be wrapped like this. Might make the code a bit more readable.
```
if (cellID <= 0) then
   cell_counter = cell_counter + 1 
   cycle
fi
```